src/hooks/useFormatGoalCalculatorConstants.ts

```diff
@@ -0,0 +1,70 @@
+import { useMemo } from 'react';
+import {
+  MpdGoalBenefitsConstantPlanEnum,
+  MpdGoalBenefitsConstantSizeEnum,
+  MpdGoalMiscConstantEnum,
+} from 'src/graphql/types.generated';
+import { GoalCalculatorConstantsQuery } from '../components/Reports/GoalCalculator/Shared/GoalCalculation.generated';
+
+export type GoalBenefitsConstantMap = Map<
+  string,
+  {
+    plan: MpdGoalBenefitsConstantPlanEnum;
+    planDisplayName: string;
+    size: MpdGoalBenefitsConstantSizeEnum;
+    sizeDisplayName: string;
+    cost: number;
+  }
+>;
+export type GoalMiscConstantMap = Map<MpdGoalMiscConstantEnum, string>;
+export type GoalGeographicConstantMap = Map<string, number>;
+
+interface FormattedConstants {
+  goalBenefitsConstantMap: GoalBenefitsConstantMap;
+  goalMiscConstantMap: GoalMiscConstantMap;
+  goalGeographicConstantMap: GoalGeographicConstantMap;
+}
+
+export const useFormatConstants = (
+  data?: GoalCalculatorConstantsQuery,
+): FormattedConstants => {
+  const goalBenefitsConstantMap: GoalBenefitsConstantMap = useMemo(() => {
+    const map = new Map();
+    data?.constant.mpdGoalBenefitsConstants?.forEach((constant) => {
+      const { plan, planDisplayName, size, sizeDisplayName, cost } = constant;
+      map.set(`${size}-${plan}`, {
+        plan,
+        planDisplayName,
+        size,
+        sizeDisplayName,
+        cost,
+      });
+    });
+    return map;
+  }, [data?.constant.mpdGoalBenefitsConstants]);
+
+  const goalMiscConstantMap = useMemo(() => {
+    const map = new Map();
+    data?.constant.mpdGoalMiscConstants?.forEach((constant) => {
+      const { category, label } = constant;
+      map.set(category, label);
+    });
+    return map;
+  }, [data?.constant.mpdGoalMiscConstants]);
+
+  const goalGeographicConstantMap = useMemo(() => {
+    const map = new Map();
```

Comment:
What do you think about writing these in functional style? This way makes it clearer to me that each constant gets its own map entry, the key is the location, and the value is the multiplier. But I'm not sure if all devs think the same way as me.

```ts
const map = new Map(data?.constant.mpdGoalGeographicConstants?.forEach((constant) => [constant.location, constant.percentageMultiplier]));
```